autorest/azure/sender.go

```diff
@@ -0,0 +1,187 @@
+package azure
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest"
+)
+
+// DoRetryForStatusCodes is the Azure specific implementation for retring requests
```

Comment:
I would update this comment to mention that it will automatically attempt to register the resource provider.

Reply:
Fixed